src/include/storage/store/table_statistics.h

```diff
@@ -15,16 +16,29 @@ using lock_t = std::unique_lock<std::mutex>;
 using atomic_uint64_vec_t = std::vector<std::atomic<uint64_t>>;
 
 class TableStatistics {
-
 public:
-    TableStatistics() : numTuples{0} {}
-
     virtual ~TableStatistics() = default;
 
-    explicit TableStatistics(uint64_t numTuples) : numTuples{numTuples} {
+    explicit TableStatistics(const catalog::TableSchema& schema) : numTuples{0} {
+        for (auto property : schema.getProperties()) {
+            propertyStatistics[property->getPropertyID()] = std::make_unique<PropertyStatistics>();
+        }
+    }
+
+    explicit TableStatistics(uint64_t numTuples,
+        std::unordered_map<common::property_id_t, std::unique_ptr<PropertyStatistics>>&&
+            propertyStatistics)
+        : numTuples{numTuples}, propertyStatistics{std::move(propertyStatistics)} {
         assert(numTuples != UINT64_MAX);
     }
 
+    explicit TableStatistics(const TableStatistics& other) : numTuples{other.numTuples} {
+        for (auto& propertyStats : other.propertyStatistics) {
+            propertyStatistics[propertyStats.first] =
+                std::make_unique<PropertyStatistics>(propertyStats.second.get());
```

Comment:
Found the issue: no copy constructor, and this is not being de-referenced anyway, so the pointer is being coerced into a bool (i.e. always true since it's not null).